app/components/syntax-highlighted-diff.ts

```diff
@@ -160,28 +177,53 @@ export default class SyntaxHighlightedDiffComponent extends Component<Signature>
     }
   }
 
-  highlightCode = task({ keepLatest: true }, async (): Promise<void> => {
-    if (this.isDarkMode === undefined) {
-      return; // Not ready to highlight yet
+  highlightCode = task({ [HIGHLIGHT_CODE_TASK_TIMEOUT_METHOD]: true }, async (): Promise<void> => {
+    // For "restartable" to have immediate effect for sequential calls
+    if (HIGHLIGHT_CODE_TASK_TIMEOUT_METHOD === 'restartable') {
+      await timeout(HIGHLIGHT_CODE_TASK_TIMEOUT_PRE);
     }
 
+    // Return if not ready to highlight yet
+    if (this.#isDarkMode === undefined) {
+      return;
+    }
+
+    // Return if nothing changed since last render
+    if (this.#isDarkMode === this.#isDarkModeRendered) {
+      return;
+    }
+
+    // Remember which mode we've last used for rendering
+    this.#isDarkModeRendered = this.#isDarkMode;
```

Comment:
Was thinking about it myself, will do! If we had error hanging then this would make sense, but we don't.

Reply:
I reworked this into the following interface and class:

```ts
interface HighlightCodeParams {
  isDarkMode?: boolean;
  language?: string;
  codeWithoutDiffMarkers?: string;
}

class HighlightCodeParamsCache implements HighlightCodeParams {
  isDarkMode?: boolean;
  language?: string;
  codeWithoutDiffMarkers?: string;

  compareWithParams({ isDarkMode, language, codeWithoutDiffMarkers }: HighlightCodeParams): boolean {
    return this.isDarkMode === isDarkMode && this.language === language && this.codeWithoutDiffMarkers === codeWithoutDiffMarkers;
  }

  setParams({ isDarkMode, language, codeWithoutDiffMarkers }: HighlightCodeParams): void {
    this.isDarkMode = isDarkMode;
    this.language = language;
    this.codeWithoutDiffMarkers = codeWithoutDiffMarkers;
  }
}
```

and now all `highlightCode` task has to do is as simple/reliable/typescript-backed as:

```ts
// Return if nothing changed since last render
if (
  this.#lastHighlightCodeParams.compareWithParams({
    isDarkMode: isDarkMode,
    language: language,
    codeWithoutDiffMarkers: codeWithoutDiffMarkers,
  })
) {
  return;
}

// ...

// Remember all parameters used for this task run
this.#lastHighlightCodeParams.setParams({
  isDarkMode: isDarkMode,
  language: language,
  codeWithoutDiffMarkers: codeWithoutDiffMarkers,
});
```

Reply:
Also the original implementation was missing a `did-update` trigger for whenever `@language` argument changed, I added it.